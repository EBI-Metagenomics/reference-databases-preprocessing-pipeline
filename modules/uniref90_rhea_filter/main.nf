process UNIREF90_RHEA_FILTER {
    label 'light'
    container 'quay.io/biocontainers/biopython:1.78'

    input:
    path uniref90_fasta
    path uniprot_rhea_mapping

    output:
    path 'uniref90_with_rhea.fasta', emit: filtered_fasta

    script:
    "uniref90_rhea_filter.py ${uniref90_fasta} ${uniprot_rhea_mapping} uniref90_with_rhea.fasta"
}