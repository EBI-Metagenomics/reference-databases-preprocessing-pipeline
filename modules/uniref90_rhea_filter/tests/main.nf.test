nextflow_process {

    name "Test Process UNIREF90_RHEA_FILTER"
    script "../main.nf"
    process "UNIREF90_RHEA_FILTER"

    test("Remove from protein fasta proteins without Rhea annotation") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta", checkIfExists: true)
                input[1] = Channel.fromPath("${moduleDir}/tests/data/uniprot_rhea_mapping.test.tsv", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
