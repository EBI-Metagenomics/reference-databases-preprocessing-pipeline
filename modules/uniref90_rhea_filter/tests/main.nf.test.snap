{
    "Remove from protein fasta proteins without Rhea annotation": {
        "content": [
            {
                "0": [
                    "uniref90_with_rhea.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "filtered_fasta": [
                    "uniref90_with_rhea.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-06T11:45:08.180244"
    }
}